janitor/functions/case_when.py

```diff
@@ -99,26 +102,51 @@ def case_when(df: pd.DataFrame, *args, column_name: str) -> pd.DataFrame:
         For a 1-D array, it should have the same length as the DataFrame.
-        The `default` argument applies if none of `condition0`,
-        `condition1`, ..., evaluates to `True`.
-        Value can be a scalar, a callable, or a 1-D array. if `default` is a
-        callable, it should evaluate to a 1-D array.
+    :param default: scalar, 1-D array or callable.
+        If callable, it should evaluate to a 1-D array.
         The 1-D array should be the same length as the DataFrame.
+        The element inserted in the output when all conditions
+        evaluate to False. Default is 0.
     :param column_name: Name of column to assign results to. A new column
         is created, if it does not already exist in the DataFrame.
-    :raises ValueError: If the condition fails to evaluate.
     :returns: A pandas DataFrame.
     """
-    conditions, targets, default = _case_when_checks(df, args, column_name)
+    check("column_name", column_name, [str])
+    default = if_else(df, args, default)
+    return df.assign(**{column_name: default})
+
+
+def if_else(df: pd.DataFrame, args, default: Optional[Any]) -> pd.Series:
+    """
+    Evaluates conditions against values;
+    useful as a standalone for use within Pandas' assign.
 
-    if len(conditions) == 1:
-        default = default.mask(conditions[0], targets[0])
-        return df.assign(**{column_name: default})
+    :param df: A pandas DataFrame.
+    :param args: Variable argument of conditions and expected values.
+        Takes the form
+        `condition0`, `value0`, `condition1`, `value1`, ... .
+        `condition` can be a 1-D boolean array, a callable, or a string.
+        If `condition` is a callable, it should evaluate
+        to a 1-D boolean array. The array should have the same length
+        as the DataFrame. If it is a string, it is computed on the dataframe,
+        via `df.eval`, and should return a 1-D boolean array.
+        `result` can be a scalar, a 1-D array, or a callable.
+        If `result` is a callable, it should evaluate to a 1-D array.
+        For a 1-D array, it should have the same length as the DataFrame.
+    :param default: scalar, 1-D array or callable.
+        If callable, it should evaluate to a 1-D array.
+        The 1-D array should be the same length as the DataFrame.
+        The element inserted in the output when all conditions
+        evaluate to False. Default is 0.
+    :raises ValueError: If the condition fails to evaluate.
+    :returns: A pandas Series.
+    """
 
+    booleans, replacements, default = _case_when_checks(df, args, default)
     # ensures value assignment is on a first come basis
-    conditions = conditions[::-1]
-    targets = targets[::-1]
-    for condition, value, index in zip(conditions, targets, count()):
+    booleans = booleans[::-1]
+    replacements = replacements[::-1]
+    for condition, value, index in zip(booleans, replacements, count()):
```

Comment:
use `enumerate` replace `itertools.count`?
So we can remove `import itertools` line.